app/src/main/java/com/rkm/tasky/network/repository/implementation/AuthorizationRepositoryImpl.kt

```diff
@@ -0,0 +1,48 @@
+package com.rkm.tasky.network.repository.implementation
+
+import com.rkm.tasky.di.IoDispatcher
+import com.rkm.tasky.network.datasource.TaskyRemoteDataSource
+import com.rkm.tasky.network.model.dto.AccessTokenDTO
+import com.rkm.tasky.network.model.request.AccessTokenRequest
+import com.rkm.tasky.network.model.response.asAccessTokenDTO
+import com.rkm.tasky.network.repository.abstraction.AuthorizationRepository
+import com.rkm.tasky.network.util.NetworkError
+import com.rkm.tasky.network.util.safeCall
+import com.rkm.tasky.util.result.EmptyResult
+import com.rkm.tasky.util.result.Result
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class AuthorizationRepositoryImpl @Inject constructor(
+    private val dataSource: TaskyRemoteDataSource,
+    @IoDispatcher private val dispatcher: CoroutineDispatcher
+): AuthorizationRepository {
```

Comment:
I'm personally not a big fan of such empty classes that just delegates calls to another class. Clean architecture is fine and all, but I personally think we need to adjust it to our project while keeping the benefits. I've seen so many projects where I had to click through 5+ classes and layers just to see what a function call does. That doesn't make code more readable, but much harder to understand.

Instead, you could just call the repo's function in your ViewModel which is totally fine, since the repo's abstraction sits in domain.

Again, just my personal view and experience. Other people might see that differently, so what you do with it is up to you :)